src/main/java/com/google/cloud/pubsublite/spark/PslSparkUtils.java

```diff
@@ -132,4 +133,25 @@ public static SparkSourceOffset getSparkStartOffset(
           "Failed to get information from PSL and construct startOffset", e);
     }
   }
+
+  // EndOffset = min(startOffset + batchOffsetRange, headOffset)
+  public static SparkSourceOffset getSparkEndOffset(
+      SparkSourceOffset headOffset,
+      SparkSourceOffset startOffset,
+      @Nullable Long maxMessagePerBatch,
```

Comment:
Don't make this nullable everywhere. Instead, use Long.MAX_VALUE as the default, and use LongMath.saturatedAdd. User provided lengths can still cause overflow.